domains/certificates/Query.API/API.IntegrationTests/Factories/QueryApiWebApplicationFactory.cs

```diff
@@ -104,7 +111,14 @@ protected override void ConfigureWebHost(IWebHostBuilder builder)
             });
 
             // Remove MeasurementsSyncerWorker
-            services.Remove(services.First(s => s.ImplementationType == typeof(MeasurementsSyncerWorker)));
+            if (!MeasurementsSyncEnabled)
+                services.Remove(services.First(s => s.ImplementationType == typeof(MeasurementsSyncerWorker)));
+
+            if (measurementsClient != null)
```

Comment:
Why is this needed?

Reply:
I made it possible to mock the client. And it is only set here if you do that